src/consumer.rs

```diff
@@ -0,0 +1,352 @@
+use crate::{SharedFluvio, DEFAULT_TOPIC, DEFAULT_PARTITION, OFFSET_BEGINNING, OFFSET_END};
+use crate::{optional_property, must_property};
+
+use log::debug;
+use flv_future_aio::task::spawn;
+use fluvio::PartitionConsumer;
+use fluvio::{Offset, FluvioError};
+use fluvio::dataplane::fetch::FetchablePartitionResponse;
+use fluvio::dataplane::record::RecordSet;
+
+use node_bindgen::derive::node_bindgen;
+use node_bindgen::core::NjError;
+use node_bindgen::core::JSValue;
+use node_bindgen::core::val::JsEnv;
+use node_bindgen::core::TryIntoJs;
+use node_bindgen::sys::napi_value;
+use node_bindgen::core::JSClass;
+use node_bindgen::core::val::JsObject;
+use node_bindgen::core::buffer::ArrayBuffer;
+
+// data corresponds to the JS event emitter `event.on('data', cb)`;
+// If this variable is changed, also need to update emitter method in TypeScript
+// index.ts file; There should be no need to change this value;
+const EVENT_EMITTER_NAME: &str = "data";
+
+const PARTITION_INDEX_KEY: &str = "partitionIndex";
+const ERROR_CODE_KEY: &str = "errorCode";
+const HIGH_WATERMARK_KEY: &str = "highWatermark";
+const LAST_STABLE_OFFSET_KEY: &str = "lastStableOffset";
+const LOG_START_OFFSET_KEY: &str = "logStartOffset";
+
+const PRODUCER_ID_KEY: &str = "producerId";
+const FIRST_OFFSET_KEY: &str = "firstOffset";
+
+const ABORTED_KEY: &str = "aborted";
+const RECORDS_KEY: &str = "records";
+
+const PARTITION_LEADER_EPOCH_KEY: &str = "partitionLeaderEpoch";
+const MAGIC_KEY: &str = "magic";
+const CRC_KEY: &str = "crc";
+const ATTRIBUTES_KEY: &str = "attributes";
+const LAST_OFFSET_DELTA_KEY: &str = "lastOffsetDelta";
+const FIRST_TIMESTAMP_KEY: &str = "firstTimestamp";
+const MAX_TIMESTAMP_KEY: &str = "maxTimeStamp";
+const PRODUCER_EPOCH_KEY: &str = "producerEpoch";
+const FIRST_SEQUENCE_KEY: &str = "firstSequence";
+
+const BATCH_OFFSET_KEY: &str = "baseOffset";
+const BATCH_LENGTH_KEY: &str = "batchLength";
+const HEADER_KEY: &str = "header";
+
+const HEADERS_KEY: &str = "headers";
+const KEY_KEY: &str = "key";
+const VALUE_KEY: &str = "value";
+
+const BATCHES_KEY: &str = "batches";
+
+pub struct PartitionConsumerWrapper {
+    client: SharedFluvio,
+    topic: String,
```

Comment:
PartitionConsumer already as topic and partition field

Reply:
commit a1cd4c8 fixes;